### PR TITLE
feat(webhooks): Add preconfigurable webhook support

### PR DESCRIPTION
**Pull request by jervi**
This PR adds support for webhook stages that are configured in the Orca and appears more or less as native stages in Deck.

skjermbilde 2017-05-09 kl 15 08 39
skjermbilde 2017-05-09 kl 15 09 55

Corresponds with https://github.com/spinnaker/orca/pull/1329 and https://github.com/spinnaker/gate/pull/389

### CODE REVIEW

File: app/scripts/modules/core/pipeline/config/stages/webhook/webhookStage.ts [hunk 1]
```diff
@@ -62,6 +79,10 @@ export class WebhookStage {
     this.stage.statusUrlResolution = this.viewState.statusUrlResolution;
   }
 
+  public customHeaderCount(): number {
+    return this.stage.customHeaders ? Object.keys(this.stage.customHeaders).length : 0;
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/pipeline/config/stages/webhook/webhookStage.ts]: Can you verify this still returns `0` when the last custom header is removed? I'm wondering if angular puts a `$$hashkey` on the `customHeaders` object when it iterates over it in the UI above, and if that gets picked up by the `Object.keys` call.

File: app/scripts/modules/core/pipeline/config/stages/webhook/webhookStage.ts [hunk 2]
```diff
@@ -78,14 +99,30 @@ export class WebhookStage {
...
   }
+
+  public displayField(field: string): boolean {
+    return !this.preconfiguredProperties || !this.preconfiguredProperties.some(property => property === field);
+  }
+
+  public noUserConfigurableFields(): boolean {
+    if (this.preconfiguredProperties) {
+      if (this.viewState.waitForCompletion) {
+        return this.preconfiguredProperties.length >= 12;
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/pipeline/config/stages/webhook/webhookStage.ts]: worth adding a comment here as to where that `12` comes from, especially if there's a chance additional properties might get added/removed in the future and we need to tweak this to keep it in sync.

**Comment by anotherchrisberry** [comment]
I really like this - nice job on the implementation. A few minor questions (and a merge conflict that I pushed on you, sorry), but LGTM otherwise.

We'll want to hold off merging this at least until the Gate PR gets merged, and probably worth waiting for Orca, too.

File: app/scripts/modules/core/pipeline/config/stages/webhook/webhookStage.ts (hunk 1 above)
Comment (jervi) [inline app/scripts/modules/core/pipeline/config/stages/webhook/webhookStage.ts]: Yes, just tested it, and it works as expected 👍

File: app/scripts/modules/core/pipeline/config/stages/webhook/webhookStage.ts (hunk 2 above)
Comment (jervi) [inline app/scripts/modules/core/pipeline/config/stages/webhook/webhookStage.ts]: Yes, I agree, it's a very magic number. I'll fix.

Comment (jervi) [inline app/scripts/modules/core/pipeline/config/stages/webhook/webhookStage.ts]: I moved the whole thing to Orca and just pass a boolean to Deck. It became much cleaner IMHO. https://github.com/spinnaker/orca/pull/1329/files#diff-bf5975fd9ad4864312a075624f079847R59

**Comment by anotherchrisberry** [comment]
Should be all set now - just needs a rebase.

**Comment by jervi** [comment]
Rebased 👍

**Comment by anotherchrisberry** [comment]
one more rebase and we'll get it merged (sorry!)

**Comment by jervi** [comment]
Hehe, yeah, that rebase lasted for a whole minute 😄

**Pull request merged**